### ISSUE TITLE
Errors do not result in non-zero exit codes

### ISSUE THREAD

**Issue by colin-nolan** [wtsi-hgi/irobot-client#6]
It doesn't seem like errors result in a non-zero exit code, making it very difficult to determine if a failure occurred, e.g.
```
$ irobotclient ...
WARNING: Could not obtain a checksum from the response ETag header.
Exiting....
$ echo $?
0
```
The code path certainly looks like this is a failure as it doesn't bother with the checksum if this problem is detected: https://github.com/wtsi-hgi/irobot-client/blob/develop/irobotclient/entrypoint.py#L84-L95.

**Comment by EWeatherill** [comment]
Good point!  Maybe it would be good better if the `_validate_download_data`exited with a non-zero code or returned an error code to be used for an exit statement higher up?
